### DISCUSSION TITLE
have option for `magit-remote-add` to default to ssh instead of https

### DISCUSSION THREAD

**Discussion by bo-tato** [magit/magit#5004] [Ideas]
99% of the time when I'm adding a remote, the upstream remote is https://... and the remote I'm adding is my fork of it. Github nolonger supports pushing to https we need to use ssh authentication, so to save me editing the remote url manually each time I simply changed `magit-add-remote`:
```
(let ((origin (string-replace "https://github.com/" "git@github.com:" (magit-get "remote.origin.url")))
      ...)
```
so that it defaults to adding the new remote as ssh authentication instead of https. I assume this would be useful in the workflow of most other people that use github, so maybe it makes sense to add as an option in magit?

**Comment by tarsius** [comment]
> 99% of the time when I'm adding a remote, the upstream remote is https://... and the remote I'm adding is my fork of it.

This is the way.

> Github nolonger supports pushing to https we need to use ssh authentication, so to save me editing the remote url manually each time I simply changed magit-add-remote: [...] so that it defaults to adding the new remote as ssh authentication instead of https.

I used to favor shh myself, but some years ago switched to using https instead. One nice effect of that is that, for most forges, you can use the same url with git and in a browser.

I don't think an option is required here, and if we added one, the suggested behavior should be opt-in, thus adding an option would not help with the discovery issue.

Git provides a nice solution for this problem, by allowing users to rewrite certain urls. For example, I have this in my `~/.config/git/config` file:

```
[url "git@github.com:"]
        insteadOf = https://github.com/
[url "git@gitlab.com:"]
        insteadOf = https://gitlab.com/
[url "git@codeberg.org:"]
        insteadOf = https://codeberg.org/
```

This modifies the urls for pushing and fetching; you could also use `pushInsteadOf`. You could further limit this to accounts that you actually have persmission to push to.

**Comment by bo-tato** [comment]
neat, thanks! I'll add that to my git config and go back to unmodified magit

**Reply by tarsius** [reply]
You're welcome!
